tex/texlive-bin/Portfile

```diff
@@ -289,6 +303,145 @@ post-destroot   {
+
+    xinstall -d ${destroot}${td_mpfd}/Doc
+    ln -s ${texlive_texmfdist}/doc ${destroot}${td_mpfd}/Doc/texmf-dist-doc
+    ln -s ${texlive_texmfports}/doc ${destroot}${td_mpfd}/Doc/texmf-doc
+    ln -s ${texlive_texmfdist}/doc ${destroot}${td_mpfd}/Doc/texmf-doc-doc
+    ln -s ${texlive_texmfdist}/doc ${destroot}${td_mpfd}/Doc/texmf-var-doc
+
+    ln -s ${prefix}/share/info ${destroot}${td_mpfd}/Info
+    ln -s ${prefix}/share/man ${destroot}${td_mpfd}/Man
+
+    # The texdist script determines if a TeX distribution is usable
+    # by testing the binaries in the directory called `uname -p`.
+    # `uname -p` outputs i386 also for x86_64 systems.
+    xinstall -d ${destroot}${td_mpfd}/Programs
+    if {[variant_isset universal]} {
+        if {[string match "*ppc*" ${configure.universal_archs}]} {
+            ln -s ${texlive_mactex_texbindir} ${destroot}${td_mpfd}/Programs/powerpc
+            ln -s ${texlive_mactex_texbindir} ${destroot}${td_mpfd}/Programs/ppc
+        }
+        if {"x86_64" in ${configure.universal_archs} || "i386" in ${configure.universal_archs}} {
+            ln -s ${texlive_mactex_texbindir} ${destroot}${td_mpfd}/Programs/i386
+        }
+        if {"x86_64" in ${configure.universal_archs}} {
+            ln -s ${texlive_mactex_texbindir} ${destroot}${td_mpfd}/Programs/x86_64
+        }
+    } else {
+        if {[string match "ppc*" ${configure.build_arch}]} {
+            ln -s ${texlive_mactex_texbindir} ${destroot}${td_mpfd}/Programs/powerpc
+            ln -s ${texlive_mactex_texbindir} ${destroot}${td_mpfd}/Programs/ppc
+        } else {
+            if {${configure.build_arch} eq "x86_64"} {
+                ln -s ${texlive_mactex_texbindir} ${destroot}${td_mpfd}/Programs/i386
+            }
+            ln -s ${texlive_mactex_texbindir} ${destroot}${td_mpfd}/Programs/${configure.build_arch}
+        }
+    }
+    ln -s ${texlive_mactex_texbindir} ${destroot}${td_mpfd}/Programs/texbin
+
+    xinstall -d ${destroot}${td_mpfd}/Resources/English.lproj
+    xinstall -m 644 ${filespath}/Description.rtf \
+        ${destroot}${td_mpfd}/Resources/English.lproj/Description.rtf
+    reinplace "s|%%PREFIX%%|${prefix}|g" \
+        ${destroot}${td_mpfd}/Resources/English.lproj/Description.rtf
+    reinplace "s|%%VERSION%%|${version}|g" \
+        ${destroot}${td_mpfd}/Resources/English.lproj/Description.rtf
+
+    ln -s ${prefix}/share ${destroot}${td_mpfd}/Root
+
+    set td_vfile [open ${destroot}${td_mpfd}/TeXDistVersion w]
+    puts ${td_vfile} "1"
+    close ${td_vfile}
+
+    ln -s ${texlive_texmflocal} ${destroot}${td_mpfd}/TexmfLocal
+    ln -s ${texlive_texmfsysvar} ${destroot}${td_mpfd}/TexmfSysVar
+}
+
+pre-activate {
+    # Delete leftover MacPorts-related TeXDist directories.
+    # See comment above in post-destroot for the rationale.
+    if {[string equal ${prefix} {/opt/local}]} {
+        if {[file exists ${td_root}/MacPorts-TeXLive.texdist]} {
```

Comment:
Could you perhaps replace this with a `for` loop? Or maybe even (pseudocode)
```
for dist in [TeXLive, teTeX]
    for f in [MacPorts-${dist}.texdist ${td_fd}/MacPorts-${dist}]
        if {[file exists ${f}]}
            delete file ${f}
```

Reply:
Done: I don't know if there is a better way of writing it in Tcl...